.pre-commit-config.yaml

```
@@ -0,0 +1,19 @@
+ci:
+  autoupdate_commit_msg: "chore: update pre-commit hooks"
+  autofix_commit_msg: "style: pre-commit fixes"
+
+repos:
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
```

```suggestion
      - id: isort
        args: [".", "--skip", "pybamm/__init__.py"]
```

Turns out, the order of imports in `pybamm`'s `__init__.py` file matters (not sure why). Also the changed made to `__init__.py` should be reverted.

---

Order matters because the expression tree classes are defined in a hierarchy, so we have to define `pybamm.Symbol` before any others. Is there a better way to do this?

---

There is a way to ignore specific import sections of a file -
```py
import a
import b
# isort: off
import x
import y
# isort: on
import f
import g
```
(ignores `x` and `y` imports)

I tried this locally, but the order of imports is also required for battery models and geometry (maybe even for other imports). We can get rid of `args` here by adding `# isort: skip_file` to the top of `__init__.py`. Not sure what is the best thing to here.

---

Can we just skip isort altogether? Why is it needed?

---

I think `isort` was added through my suggestions, as it is a standard pre-commit hook 😅. Can be removed if not needed!